packages/flutter_driver/lib/src/retry.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+/// Performs an action and returns either the result of the action or a [Future]
+/// that evaluates to the result.
+typedef dynamic Action();
+
+/// Performs [action] repeatedly until it either succeeds or [timeout] limit is
+/// reached.
+///
+/// When the retry time out, the last seen error and stack trace are returned in
+/// an error [Future].
+Future<dynamic> retry(Action action, Duration timeout,
+    Duration pauseBetweenRetries) async {
```

Comment:
Tried with newlines on both sides. The current format looks better. Leaving as is.